### PR TITLE
Read all device states at once for raw input handlers

### PR DESCRIPTION
**Pull request by ekrctb**
Use the new API of <s>ppy/opentk#25</s> -> ppy/opentk#42.
- Closes ppy/osu#3073
---

### CODE REVIEW

**Comment by peppy** [comment]
Blocking this pending benchmarking for potential increases in allocs.

File: osu.Framework/Input/Handlers/Joystick/OsuTKJoystickHandler.cs [hunk 1]
```diff
@@ -72,35 +71,28 @@ private void handleState(JoystickDevice device, JoystickState newState)
...
+            for (int index = 0; index < rawStates.Count; index++)
+            {
+                var rawState = rawStates[index];
+                if (index < devices.Count)
                 {
-                    mostSeenDevices--;
-                    if (dev.LastState != null)
-                        handleState(dev, new JoystickState());
-
-                    devices.RemoveAt(i--);
```
Comment (peppy) [inline osu.Framework/Input/Handlers/Joystick/OsuTKJoystickHandler.cs]: is there a reason we don't handle disconnected devices any more?

Comment (ekrctb) [reply osu.Framework/Input/Handlers/Joystick/OsuTKJoystickHandler.cs]: I'm not entirely sure but probably disconnected devices are skipped for update.

Comment (peppy) [reply osu.Framework/Input/Handlers/Joystick/OsuTKJoystickHandler.cs]: I guess the weird case would be if a new device was connected and assume the index of a previous one. But this should still handle OK as long as we don't identify individual devices, right?

Comment (ekrctb) [reply osu.Framework/Input/Handlers/Joystick/OsuTKJoystickHandler.cs]: I'm a bit confused but,
1. OpenTK doesn't reassign new device to previously disconnected device index IIRC (not entirely sure part is here).
2. Even if did, no bad thing should happen.
3. We don't have disconnected devices most of the time and linear time is wasted for each such device anyway so no performance concerns are there.

**Pull request merged**